### PR TITLE
plugins/explain.py: Added explanation for "do not ask to ask".

### PR DESCRIPTION
**Pull request by yashasingh**
This patch adds response in case a user asks abrupt question.

Closes https://github.com/coala/corobo/issues/389

### CODE REVIEW

**Comment by yashasingh** [comment]
I have made the required changes.

File: plugins/explain.py [hunk 1]
```diff
@@ -83,6 +83,16 @@ class Explain(BotPlugin):
                      'newcomer Pull Request\n- merge a difficulty/low Pull '
                      'Request\n- review at least a difficulty/low or higher '
                      'Pull Request'
+        'do not ask to ask': 'This is a lazy way of asking your question '
+                             'because  you\'re asking for more than what '
+                             'you think you\'re asking. The solution is not '
+                             'to ask to ask, but just to ask by actaully '
+                             'formalizing it in words. Someone who is idling '
+                             'on the channel and only every now and then '
+                             'glances what\'s going on is unlikely to answer '
```
Comment (andrewda) [inline plugins/explain.py]: "glances __at__ what's going on [...]"

File: plugins/explain.py [hunk 2]
```diff
@@ -83,6 +83,16 @@ class Explain(BotPlugin):
                      'newcomer Pull Request\n- merge a difficulty/low Pull '
                      'Request\n- review at least a difficulty/low or higher '
                      'Pull Request'
+        'do not ask to ask': 'This is a lazy way of asking your question '
+                             'because  you\'re asking for more than what '
+                             'you think you\'re asking. The solution is not '
+                             'to ask to ask, but just to ask by actaully '
+                             'formalizing it in words. Someone who is idling '
+                             'on the channel and only every now and then '
```
Comment (andrewda) [inline plugins/explain.py]: "and only **occasionally** glances [...]" sounds a bit more natural here. Otherwise the "and"s sound a little weird.

Comment (yashasingh) [reply plugins/explain.py]: Okay. I shall do this.

File: plugins/explain.py [hunk 3]
```diff
@@ -83,6 +83,16 @@ class Explain(BotPlugin):
                      'newcomer Pull Request\n- merge a difficulty/low Pull '
                      'Request\n- review at least a difficulty/low or higher '
                      'Pull Request'
+        'do not ask to ask': 'This is a lazy way of asking your question '
```
Comment (prnvdixit) [inline plugins/explain.py]: Please, rephrase this line to be less "You did something quite wrong" to more "It's Ok, you just need to be careful from now on and ask specific questions". Current language might scare newcomers away 😄.

File: plugins/explain.py [hunk 4]
```diff
@@ -83,6 +83,16 @@ class Explain(BotPlugin):
...
+                             'because  you\'re asking for more than what '
+                             'you think you\'re asking. The solution is not '
+                             'to ask to ask, but just to ask by actaully '
```
Comment (prnvdixit) [inline plugins/explain.py]: You haven't explained what __don't "ask to ask"__ actually means in a proper way (like describe the specific problem, tell us what you did to solve and didn't worked out, what others might have to do to regenerate the problem) :wink:

**Comment by LifetimeAlone** [comment]
Your branch is out-of-date, try to rebase it (https://coala.io/rebase)
It's a bit better to say 'Fixes' instead of 'Closes' in the commit message, as the issue is labeled 'bug' :wink:

**Comment by Makman2** [comment]
@gitmate-bot rebase

**Comment by Makman2** [comment]
Why is gitmate not complaining about the commit... it violates our style rules... CC @sils

**Comment by Makman2** [comment]
However, this is my local coala run on this PR ;)
```
Project wide:
|    | [NORMAL] GitCommitBear:
|    | Shortlog of HEAD commit isn't in imperative mood! Bad words are 'Added'

Project wide:
|    | [NORMAL] GitCommitBear:
|    | Shortlog of the HEAD commit contains 59 character(s). This is 9 character(s) longer than the limit (59 > 50).
```

**Comment by Makman2** [comment]
unack dbe798a

**Comment by yashasingh** [comment]
@Makman2 Should I change my commit message ? Is there something else also to be done ?

**Comment by prnvdixit** [comment]
@yashasingh Just go through https://coala.io/commit - You would be fine 😄

File: plugins/explain.py [hunk 5]
```diff
@@ -82,7 +82,19 @@ class Explain(BotPlugin):
...
                      'newcomer Pull Request\n- merge a difficulty/low Pull '
                      'Request\n- review at least a difficulty/low or higher '
-                     'Pull Request'
+                     'Pull Request',
+        'do not ask to ask': 'This seems a lazy way of asking your question '
+                             'Don\'t take it personally. It\'s just that '
+                             'you\'re asking for more than what you think '
+                             'you\'re asking. The solution is not to simply '
+                             'question people about them, but just to '
+                             'ask for your piece of information by actaully '
```
Comment (Mixih) [inline plugins/explain.py]: typo in "actually"

File: plugins/explain.py [hunk 6]
```diff
@@ -82,7 +82,19 @@ class Explain(BotPlugin):
                      'newcomer Pull Request\n- review at least a difficulty/'
                      'newcomer Pull Request\n- merge a difficulty/low Pull '
                      'Request\n- review at least a difficulty/low or higher '
-                     'Pull Request'
+                     'Pull Request',
+        'do not ask to ask': 'This seems a lazy way of asking your question '
+                             'Don\'t take it personally. It\'s just that '
```
Comment (Mixih) [inline plugins/explain.py]: use double quotes around the whole string instead of escaping the single quotes. That way you can keep the single quotes in the string more readable. (this comment applies to all of the instances)

File: plugins/explain.py [hunk 7]
```diff
@@ -82,7 +82,19 @@ class Explain(BotPlugin):
...
+                             'you\'re asking for more than what you think '
+                             'you\'re asking. The solution is not to simply '
```
Comment (Mixih) [inline plugins/explain.py]: the sentance `The solution is not to simply [snip]...` is too wordy. Please reword it to be more concise.

File: plugins/explain.py [hunk 8]
```diff
@@ -82,7 +82,19 @@ class Explain(BotPlugin):
...
+                             'question people about them, but just to '
+                             'ask for your piece of information by actaully '
+                             'formalizing it in words. Someone who is idling '
+                             'on the channel and only occasionally glances '
+                             'at what\'s going on is unlikely to answer '
+                             'to your "asking to ask" question, but your '
```
Comment (Mixih) [inline plugins/explain.py]: delete `to`

**Comment by jayvdb** [comment]
unack https://github.com/coala/corobo/pull/404/commits/47c1fd04c996c351a583a2ee8bf611783a4efaaf

**Comment by prnvdixit** [comment]
Also, commit Shortlog should be imperative - "Explain...."
